pkg/cloud/services/compute/instance.go

```diff
@@ -107,13 +107,19 @@ func (s *Service) createInstanceImpl(eventObject runtime.Object, instanceSpec *I
 		BlockDevice:      blockDevices,
 	}
 
+	schedulerAdditionalProperties := make(map[string]interface{})
+	for k, v := range instanceSpec.SchedulerAdditionalProperties {
+		schedulerAdditionalProperties[k] = v
+	}
+
 	server, err := s.getComputeClient().CreateServer(
 		keypairs.CreateOptsExt{
 			CreateOptsBuilder: serverCreateOpts,
 			KeyName:           instanceSpec.SSHKeyName,
 		},
 		servers.SchedulerHintOpts{
-			Group: instanceSpec.ServerGroupID,
+			Group:                instanceSpec.ServerGroupID,
+			AdditionalProperties: schedulerAdditionalProperties,
```

Comment:
and as group is part of the scheduler properties ,should we consider consolidate them?

Reply:
@jichenjc well, it is true that all those fields in `servers.schedulerHintOpts` are merged and passed as `os:scheduler_hints` request body parameter, so we can set `group` under `additionalProperties`. But I am not sure if that is necessary because gophercloud already defined `Group` as an explicit parameter (i think it is just for convenient) under `servers.SchedulerHintOpts`. 
https://github.com/gophercloud/gophercloud/blob/5cb81d730a1e027aa5981edd23c7b1403411db40/openstack/compute/v2/servers/requests.go#L141

Reply:
I think it's correct to leave Group separate here. I considered asking that we validate that AdditionalProperties does not contain `group`, but the problem with that is that we can't easily extend that in the future as we add new 'supported' hints.

So I think additionalProperties is a footgun, but there's no good way round that. We should just document that adding 'group' to it is a bad idea and has undefined behaviour.